modules/stochastic_tools/src/surrogates/NearestPointSurrogate.C

```diff
@@ -22,19 +22,37 @@ NearestPointSurrogate::validParams()
 
 NearestPointSurrogate::NearestPointSurrogate(const InputParameters & parameters)
   : SurrogateModel(parameters),
-    _sample_points(getModelData<std::vector<std::vector<Real>>>("_sample_points"))
+    _sample_points(getModelData<std::vector<std::vector<Real>>>("_sample_points")),
+    _sample_results(getModelData<std::vector<std::vector<Real>>>("_sample_results"))
 {
 }
 
 Real
 NearestPointSurrogate::evaluate(const std::vector<Real> & x) const
 {
   // Check whether input point has same dimensionality as training data
-  mooseAssert((_sample_points.size() - 1) == x.size(),
+  mooseAssert(_sample_points.size() == x.size(),
               "Input point does not match dimensionality of training data.");
 
-  // Returned value from training data (first sample is default)
-  Real val = _sample_points.back()[0];
+  return findNearestPoint(x, 0);
+}
+
+void
+NearestPointSurrogate::evaluate(const std::vector<Real> & x, std::vector<Real> & y) const
+{
+  mooseAssert(_sample_points.size() == x.size(),
+              "Input point does not match dimensionality of training data.");
+
+  y.assign(_sample_results.size(), 0.0);
+
+  for (const auto & r : make_range(y.size()))
+    y[r] = findNearestPoint(x, r);
+}
+
+Real
+NearestPointSurrogate::findNearestPoint(const std::vector<Real> & x, const unsigned int r) const
```

Comment:
Let's make this slightly more efficient. Instead of returning the value, we should return the index in `_sample_points` then the `evaluate` functions can just use it to into `_sample_results`.